### PR TITLE
feat(curriculum): add solution

### PR DESCRIPTION
**Pull request by gikf**
- Adds solution
- Adds definition of `_testTriangle` in the background, so it's not needed to include it in solution.
- Changes tests to use both test cases defined from the background.

Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

### CODE REVIEW

File: curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-67-maximum-path-sum-ii.md [hunk 1]
```diff
@@ -28,27 +28,31 @@ Find the maximum total from top to bottom in `numTriangle`, a 2D array defined i
 `maximumPathSumII(testTriangle)` should return a number.
 
 ```js
-assert(typeof maximumPathSumII(testTriangle) === 'number');
+assert(typeof maximumPathSumII(_testTriangle) === 'number');
```
Comment (nhcarrigan) [inline curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-67-maximum-path-sum-ii.md:31]: Why the change in variable names?

Comment (gikf) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-67-maximum-path-sum-ii.md:31]: I wanted to avoid any possible conflict with the Camper code and variables defined in background. Although I believe test runner would handle having the same name defined twice, in the Camper code and background, without errors, in one of previously changed challenges it was suggested to still avoid using the same name twice.

Comment (ojeytonwilliams) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-67-maximum-path-sum-ii.md:31]: It would clash, unfortunately.  You can test this by using
```
const testTriangle = [[6, 0, 0, 1],
                      [7, 4, 0, 0],
                      [2, 4, 6, 0],
                      [8, 5, 9, 3]];
```
in the challenge right now in production.  Even if the algorithm is correct, the tests will fail.

The test runner will not throw errors in this case, it will just silently replace any variables defined in `--before-user-code--` with the user's definitions.

Comment (gikf) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-67-maximum-path-sum-ii.md:31]: In production that would fail, because currently `testTriangle` is defined only in seed code :stuck_out_tongue:. 

Either way, despite of no explicit errors caused by repeated names, I get there's doze of surprising behavior with it, so it's best to avoid it all together.

Comment (ojeytonwilliams) [reply curriculum/challenges/english/10-coding-interview-prep/project-euler/problem-67-maximum-path-sum-ii.md:31]: Ooof, good point.  What I was saying applies to `numTriangle`, at least. You can replace it the user code and break the tests.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1:

**Review by naomi-lgbt — APPROVED** [review]

**Comment by moT01** [comment]
I added this solution to the forum guide article
